File: raspberry/websocket_command.py
from dataclasses import dataclass
from enum import Enum

class WebsocketCommandType(Enum):
  MOTOR = 1

@dataclass
class WebsocketCommand:
  type: WebsocketCommandType
  x: int
  y: int
  cutter: bool

  def __post_init__(self):
      self.type = WebsocketCommandType[self.type]

File: mobile/lib/websocket_command.dart
import 'dart:convert' as convert;

enum WebsocketCommandType {
  MOTOR,
}

class WebsocketCommand {
  final WebsocketCommandType type;
  final int x;
  final int y;
  final bool cutter;

  const WebsocketCommand({
    required this.type,
    required this.x,
    required this.y,
    required this.cutter,
  });

  String toJson() {
    return convert.jsonEncode({
      'type': type.name.toUpperCase(),
      'x': x,
      'y': y,
      'cutter': cutter,
    });
  }
}